docs/guide/philosophy.md

```diff
@@ -0,0 +1,21 @@
+
+## A Pragmatic Approach to Performance
+
+Vite has been focused on performance since its [origins](./why.md). Its dev server architecture allows HMR that stays fast as projects scale. Vite uses native tools like [esbuild](https://esbuild.github.io/) and [SWC](https://github.com/vitejs/vite-plugin-react-swc) to implement intensive tasks but keeps the rest of the code in JS to balance speed with flexibility. When needed, framework plugins will tap into [Babel](https://babeljs.io/) to compile user code. And during build time Vite currently uses [Rollup](https://rollupjs.org/) where bundling size and having access to a wide ecosystem of plugins are more important than raw speed. Vite will continue to evolve internally, using new libraries as they appear to improve DX while keeping its API stable.
+
+## Building Frameworks on top of Vite
+
+Although Vite can be used by users directly, it shines as a tool to create frameworks. Vite core is framework agnostic, but there are polished plugins for each UI frameworks. Its [JS API](./api-javascript.md) allows App Framework authors to use Vite features to create tailored experiences for their users. Vite includes support for [SSR primitives](./ssr.md), usually present in higher-level tools but fundamental to building modern web frameworks. And Vite plugins complete the picture by offering a way to share between frameworks. Vite is also a great fit when paired with [Backend frameworks](./backend-integration.md) like [Ruby](https://vite-ruby.netlify.app/) and [Laravel](https://laravel.com/docs/10.x/vite).
+
+## Vite-powered Tooling
+
+The Vite Ecosystem works to ensure Vite projects can seamlessly integrate with the wide range of tools we use to develop modern apps. For example, Vite has first-class support in [Storybook](https://storybook.js.org/docs/react/builders/vite). Vite is also used to power new features in common tools. Vite powers Component Testing in both [Playwright](https://playwright.dev/docs/test-components) and [Cypress](https://docs.cypress.io/guides/component-testing/overview). The ecosystem tries to work with established projects when possible, but there are cases where Vite enables the creation of new projects as in the case of [Vitest](https://vitest.dev/), a fast test framework. A common theme in tools based on Vite is the collapsing of configuration for a project under a unified Vite pipeline. A single `vite.config.js` would be used to configure the build, dev, and test environments.
```

Comment:
I don't quite understand this section as a philosophy. Tools that use Vite commonly reuse `vite.config.js`, but it isn't something we enforce as a philosphy, but something that the tooling chose to lean into.

Or perhaps I'm missing something here.

Reply:
The part about sharing a single config for different env/tools was a detail. 
I think the main point of this section was to show there is an active effort to ensure that other tools work well with Vite. But I don't think it works, I'll delete it for now.